youtube_dl/extractor/cbs.py

```diff
@@ -20,6 +24,101 @@ def _parse_smil_subtitles(self, smil, namespace=None, subtitles_lang='en'):
+    IE_DESC = 'CBS show playlists, including full episodes and clips'
+    IE_NAME = 'cbs.com:playlist'
+    _VALID_URL = r'(?i)https?://(?:www\.)cbs.com/shows/(?P<id>[\w-]+)/?$'
+    _TEST = {
+        'url': 'http://www.cbs.com/shows/the-late-show-with-stephen-colbert',
+        'info_dict': {
+            'id': 61456254,
+            'title': 'The Late Show with Stephen Colbert',
+        },
+        'playlist_mincount': 14,
+        # If uncommented, the test harness tries to download all 30 playlist entries.
+        # Even limited to 10KB each, this can take 15 minutes. Not reasonable.
+        # 'playlist': [{
+        #     'info_dict': {
+        #         'id': 'xxx',
+        #         'ext': 'xxx.mp4',
+        #         },
+        # }],
+    }
+
+    def carousel_playlist(self, url, type):
+        carousel = self._download_json(url, 'Downloading %s carousel' % type)
+        episodes = carousel['result']['data']
+        carousel_title = episodes[0]['series_title']
+
+        entries = []
+        for ep in episodes:
+            entries.append(self.url_result(
+                urljoin(url, ep['app_url']),
+                'CBS',
+                ep['content_id'],
+                ep['episode_title']))
+
+        return self.playlist_result(entries, playlist_title=carousel_title)
+
+    def _real_extract(self, url):
+        show_name = self._match_id(url)
+        webpage = self._download_webpage(url, show_name)
+
+        # not-quite JSON, no double-quotes:
+        #  var show = new CBS.Show({id:61456254});
+        show_id_json = self._search_regex(r'new CBS\.Show\(([^)]*)\);', webpage, 'show_id')
+
+        show = self._parse_json(show_id_json, show_name, transform_source=js_to_json)
+
+        # Found in http://www.cbs.com/assets/min/js/min/com.cbs.min.js?20170303-224247
+        # unminified at http://www.cbs.com/assets/js/min/com.cbs.js
+        # http://www.cbs.com/carousels/shows/61456254/offset/0/limit/15/xs/0/
+        # => {id: 240172, title: "Full Episodes",
+        episodes_url = urljoin(url, '/carousels/shows/%d/offset/0/limit/15/xs/0/' % show['id'])
+
+        #  var loader = new CBS.V2.CarouselLoader({
+        #                        'video-preview-carousel': function(element) {
+        #                            element.videoCarousel({
+        #                                id          : 241426,
+        #                                templates   : 'carousels/videoAdaptive',
+        #                                scroll      : 3,
+        #                                layout      : 3,
+        #                                start       : 0,
+        #                                saveState   : false
+        #                            });
+        #                        }
+        try:
+            clipdata = self._parse_json(
+                self._search_regex(r'element\.videoCarousel\(([^)]*)\);', webpage,
+                                   'clip carousel'),
```

Comment:
If I used default, then I would have to have an `if` test for the default condition and then skip the `urljoin()` and `carousel_playlist()` calls. Is that really better? It seems to me that it would be worse.